components/shapediver/parameter/ParameterSelectionComponent.tsx

```diff
@@ -192,13 +201,47 @@ export default function ParameterSelectionComponent(
 		}
 	}, [state.uiValue]);
 
+	/**
+	 * Callback function to cancel the selection.
+	 * It resets the selection to the last value and ends the selection process.
+	 */
+	const cancel = useCallback(() => {
+		resetSelection(value);
+	}, [resetSelection, value]);
+
 	/**
 	 * Callback function to clear the selection.
 	 */
 	const clearSelection = useCallback(() => {
 		setSelectedNodeNamesAndRestoreSelection([]);
 	}, []);
 
+	/**
+	 * Effect to manage the interaction request for the selection.
+	 * It adds an interaction request when the selection is active and removes it when the selection is inactive.
+	 * It also cleans up the interaction request when the component is unmounted or when the selection state changes.
+	 */
+	useEffect(() => {
+		if (selectionActive && !interactionRequestToken) {
+			const returnedToken = addInteractionRequest({
+				type: "active",
+				viewportId,
+				disable: cancel,
+			});
+			setInteractionRequestToken(returnedToken);
+		} else if (!selectionActive && interactionRequestToken) {
+			removeInteractionRequest(interactionRequestToken);
+			setInteractionRequestToken(undefined);
+		}
+
+		return () => {
+			if (interactionRequestToken) {
+				removeInteractionRequest(interactionRequestToken);
+				setInteractionRequestToken(undefined);
+			}
+		};
+	}, [selectionActive, interactionRequestToken, cancel]);
+
```

Comment:
@MajorMeerkatThe3rd it would probably be more efficient (and easier to read, less code) if we use a ref instead of a state for `interactionRequestToken`. Otherwise the useEffect hook will always be executed twice.